### PR TITLE
Feature/2128 Generate and process pupil census CSV

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
- Script to generate schools with configurable length `node generate-school-records.js -s 200`
- Script to generate pupil census csv with configurable length  `node generate-census-records.js -r 5000`
- Read csv
-  Bulk data submission
- Optionally provide parameter to omit school lookup from timer `node index.js -l`

### CODE REVIEW

File: spike-worker/generate-census-records.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  }
+  const headers = ['LEA', 'Estab', 'UPN', 'Surname', 'Forename', 'Middlenames', 'Gender', 'DOB']
+  const ws = fs.createWriteStream(outputFilename, { flags: 'a' })
+  csv
+    .write(data, {headers: headers})
+    .pipe(ws)
+  ws.write('\n')
+}
+
+async function main (options) {
```
Comment (jon-shipley) [inline spike-worker/generate-census-records.js]: Does this need to be async?  I don't see any `await`s

Comment (GeorgeChatzigiannis) [reply spike-worker/generate-census-records.js]: @jon-shipley I forgot to remove it, thanks.

File: spike-worker/bulkImport.js [hunk 2]
```diff
@@ -35,14 +37,30 @@ const doBulkInsert = (csvPayload) => {
...
+  // Fetch all school ids for pupil records
+  const schoolDfeNumbers = csvPayload.map(r => `${r[0]}${r[1]}`)
+  const schools = await schoolDataService.sqlFindByDfeNumbers(schoolDfeNumbers)
 
   for (let index = 0; index < csvPayload.length; index++) {
     const csvRow = csvPayload[index]
     // TODO map all fields into the object
-    bulkLoad.addRow({ firstName: csvRow.firstName, lastName: csvRow.lastName })
+    bulkLoad.addRow({
+      school_id: schools.find(s => s.dfeNumber === parseInt(`${csvRow[0]}${csvRow[1]}`)),
```
Comment (GuyHarwood) [inline spike-worker/bulkImport.js]: would it not be best to run through this first and establish if any of the schools in the csv file do not exist in the database?

Comment (GeorgeChatzigiannis) [reply spike-worker/bulkImport.js]: @GuyHarwood indeed, I have updated that part.

**Pull request merged**